packages/volto/src/hooks/user/useUser.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import jwtDecode from 'jwt-decode';
+import { getUser } from '@plone/volto/actions';
+
+const useFetchUser = () => {
+  const user = useSelector((state) => state.users?.user);
+  const userId = useSelector((state) =>
+    state.userSession.token ? jwtDecode(state.userSession.token).sub : '',
+  );
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!user?.id && userId) {
```

Comment:
Rather than checking user?.id, I think we should check state.users?.get which should exist as soon as getUser is dispatched the first time (including while the request is in progress)

Reply:
`state.users?.get` exists even when `user?.id` does not exist.

It looks like this:

```javascript
{
    "error": null,
    "loaded": false,
    "loading": false
}
```
I think this if would work:

```javascript
if (!user?.id && users?.get.loading === false) {
```

Reply:
Done